homeassistant/components/supla/light.py

```diff
@@ -0,0 +1,175 @@
+
+from . import DOMAIN, SUPLA_COORDINATORS, SUPLA_SERVERS
+from .entity import SuplaEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPLA_DIMMER = "DIMMER"
+SUPLA_RGBLIGHTING = "RGBLIGHTING"
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Supla lights."""
+    if discovery_info is None:
+        return
+
+    _LOGGER.debug("Discovery: %s", pformat(discovery_info))
+
+    entities: list[LightEntity] = []
+    for device in discovery_info.values():
+        device_name = device["function_name"]
+        server_name = device["server_name"]
+
+        if device_name == SUPLA_DIMMER:
+            entities.append(
+                SuplaDimmerEntity(
+                    device,
+                    hass.data[DOMAIN][SUPLA_SERVERS][server_name],
+                    hass.data[DOMAIN][SUPLA_COORDINATORS][server_name],
+                )
+            )
+
+        elif device_name == SUPLA_RGBLIGHTING:
+            entities.append(
+                SuplaRGBLightingEntity(
+                    device,
+                    hass.data[DOMAIN][SUPLA_SERVERS][server_name],
+                    hass.data[DOMAIN][SUPLA_COORDINATORS][server_name],
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class SuplaDimmerEntity(SuplaEntity, LightEntity):
+    """Representation of a Supla dimmable light."""
+
+    _attr_color_mode = ColorMode.BRIGHTNESS
+    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the switch."""
+        if ATTR_BRIGHTNESS in kwargs:
+            # Convert from Home Assistant brightness (0-255) to (0-100)
+            brightness = min(100, ceil(kwargs[ATTR_BRIGHTNESS] * 100 / 255.0))
+            await self.async_action("SET_RGBW_PARAMETERS", brightness=brightness)
+            return
+        await self.async_action("TURN_ON")
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the switch."""
+        await self.async_action("TURN_OFF")
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if switch is on."""
+        if state := self.channel_data.get("state"):
+            return state["on"]
+        return False
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of the light."""
+        state = self.channel_data.get("state")
+        if not state or "brightness" not in state:
+            return None
+        brightness = state.get("brightness")
+        # Convert from (0-100) to Home Assistant brightness (0-255)
+        return min(255, ceil(brightness * 255 / 100.0))
+
+
+class SuplaRGBLightingEntity(SuplaEntity, LightEntity):
+    """Representation of a Supla RGB light."""
+
+    _attr_color_mode = ColorMode.HS
+    _attr_supported_color_modes = {ColorMode.HS}
```

Comment:
How come `ColorMode.RGB` is not used instead to avoid the conversions back and forth between HS(V) and RGB?

Reply:
Good question. I thought that externally this could be a HS entity. I can't remember why now, but perhaps it somehow worked better/more elegant in HA?